### PR TITLE
feat: Add String Concatenation to String Interpolation Fixer

### PR DESCRIPTION
**Pull request by jlecordier**
Solves #2334

### CODE REVIEW

File: src/Fixer/Operator/StringConcatenationToInterpolationFixer.php [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+/**
+ * @author Jean LECORDIER <jean.lecordier@infoclimat.fr>
+ */
+final class StringConcatenationToInterpolationFixer extends AbstractFixer
+{
+    public function getName(): string
+    {
+        return 'Infoclimat/string_concatenation_to_interpolation';
+    }
+
```
Comment (Wirone) [inline src/Fixer/Operator/StringConcatenationToInterpolationFixer.php:33]:
```suggestion
```

This have to be removed, built-in fixers have different naming convention.

**Comment by jlecordier** [comment]
Oups, forgot to remove my custom vendor name

File: src/Fixer/Operator/StringConcatenationToInterpolationFixer.php (hunk 1 above)
Comment (jlecordier) [reply src/Fixer/Operator/StringConcatenationToInterpolationFixer.php]: Yup, just seen that myself, sorry for that

**Comment by Wirone** [comment]
@jlecordier please mark this as ready for review when CI is green. You can run `composer qa` locally, it must pass 100% 🙂.

**Comment by jlecordier** [comment]
I can't use functions ? I must put everything in the Fixer class ?

+----------------------------------------+--------------------+
| Unknown Symbol                         | Guessed Dependency |
+----------------------------------------+--------------------+
| get_close_curly_token                  |                    |
| get_close_interpolation_token          |                    |
| get_closing_array_access_index         |                    |
| get_closing_object_access_index        |                    |
| get_closing_quote_index                |                    |
| get_interpolation_boundary_token       |                    |
| get_open_curly_token                   |                    |
| get_open_interpolation_token           |                    |
| get_raw_string_token                   |                    |
| get_T_CONSTANT_ENCAPSED_STRING_content |                    |
| get_variable_end_index                 |                    |
| is_array_access_opening                |                    |
| is_concatenation                       |                    |
| is_empty_string                        |                    |
| is_interpolation_quote                 |                    |
| is_method_opening                      |                    |
| is_object_access_opening               |                    |
| is_simple_string                       |                    |
| is_variable                            |                    |
+----------------------------------------+--------------------+

**Comment by jlecordier** [comment]
I'm wondering if my function `get_variable_end_index` could be added to the `Tokens` class ? I think it must be often used ?

**Comment by Wirone** [comment]
> I can't use functions ? I must put everything in the Fixer class ?

Not necessarily in the fixer's class, if this is something to be reused across several fixers it can be added to analyzers or other classes. We rather don't do functional programming in this project. Lack of DI makes it somehow harder to reuse the code in sane way, but it's still doable.

**Comment by jlecordier** [comment]
> Lack of DI makes it somehow harder to reuse the code in sane way, but it's still doable.

I don't understand what you mean ? My function are pure and do not need DI, what are you referring to ?

And what is doable ? Functional programming ? If yes, how ? I can't pass `composer qa` because of the functions.

What is the practice that I should follow to transform my function into method ? 
Should I create regular instance method ? Or static method ?
For now, I will put everything into my Fixer class, not to pollute the other files, and it could be done in an other step.

**Comment by jlecordier** [comment]
TODO : This rule must take into account type casting

Example : `(cast) $var . ' string'` must NOT turn into `(cast) "{$var} string"`

**Comment by jlecordier** [comment]
- [ ] TODO: Check handling of `\n`

**Comment by jlecordier** [comment]
> @jlecordier instead of adding comments with todos, just add test cases - they can fail for now, and PR won't get merged until these are covered.
> 
> For other todos it's better to edit PR's description and keep the checklist there 🙂.

Thanks for the suggestions but can't I have some peace, even on my own PR lmao ? 😭

**Comment by Wirone** [comment]
> Thanks for the suggestions but can't I have some peace, even on my own PR lmao ? 😭

I get notification for your every self-comment, while editing the description would provide the same outcome (checklist) without triggering notifications 😉. So it's not about enforcing some convention on you, but I just take care about myself and other people with notifications for project's every activity 😅.

**Pull request closed without merging**